scripts/generate_conda_file.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/python
+
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# This script creates yaml files to build conda environments
+# For generating a conda file for running only python code:
+# $ python generate_conda_file.py
+# For generating a conda file for running python gpu:
+# $ python generate_conda_file.py --gpu
+# For generating a conda file for running pyspark:
+# $ python generate_conda_file.py --pyspark
+# For generating a conda file for running python gpu and pyspark:
+# $ python generate_conda_file.py --gpu --pyspark
+# For generating a conda file for running python gpu and pyspark with a particular version:
+# $ python generate_conda_file.py --gpu --pyspark-version 2.4.0
+
+import argparse
+import textwrap
+
+
+HELP_MSG = """
+To create the conda environment:
+$ conda env create -f {conda_env}.yaml
+
+To update the conda environment:
+$ conda env update -f {conda_env}.yaml
+
+To register the conda environment in Jupyter:
+$ conda activate {conda_env}
+$ python -m ipykernel install --user --name {conda_env} --display-name "Python ({conda_env})
+"""
+
+CHANNELS = ["conda-forge", "pytorch", "fastai", "defaults"]
+
+CONDA_BASE = {
+    "dask": "dask>=0.17.1",
+    "fastai": "fastai>=1.0.40",
+    "fastparquet": "fastparquet>=0.1.6",
+    "gitpython": "gitpython>=2.1.8",
+    "ipykernel": "ipykernel>=4.6.1",
+    "jupyter": "jupyter>=1.0.0",
+    "matplotlib": "matplotlib>=2.2.2",
+    "numpy": "numpy>=1.13.3",
+    "pandas": "pandas>=0.23.4",
+    "pymongo": "pymongo>=3.6.1",
+    "python": "python==3.6.8",
+    "pytest": "pytest>=3.6.4",
+    "seaborn": "seaborn>=0.8.1",
+    "scikit-learn": "scikit-learn==0.19.1",
+    "scipy": "scipy>=1.0.0",
+    "scikit-surprise": "scikit-surprise>=1.0.6",
+    "tensorflow": "tensorflow==1.12.0",
```

Comment:
I think TF tends to be behind on conda:
https://anaconda.org/conda-forge/tensorflow

Reply:
Better to have TF under pip.

Reply:
Also another library (probably azureml) pip installs tensorflow anyway.

Reply:
@anargyri why is better to have TF under pip instead of conda?

Reply:
Because we currently have
`tensorflow 1.12.0 <pip>`
`tensorflow 1.5.0 py36_0 conda-forge`
in the environment, so the conda version is way behind the pip one.